src/adapter/src/catalog.rs

```diff
@@ -6899,9 +6899,11 @@ impl Catalog {
         PersistParameters {
             blob_target_size: Some(config.persist_blob_target_size()),
             blob_cache_mem_limit_bytes: Some(config.persist_blob_cache_mem_limit_bytes()),
+            compaction_enabled: Some(config.persist_compaction_enabled()),
```

Comment:
This is sort of a weird case, since our internal guide suggests both:
- Prefixing the flag with `enable_`. (Not quite the naming scheme here, but almost!)
- Using `false` as the default, and `true` as the override value.

Which makes sense for a rollout-style flag, but we can't really do both here. (A previous job would call this sort of flag - where it's never meant to be fully enabled, only during testing or incidents - as a "circuit breaker".)

My personal preference would be to have these as `disable_{compaction,gc}` and make the default `false`. But if we go the other way, it might be slightly more standard to rename the flags to `enable_gc`...